src/memspaces/memspace_host.c

```diff
@@ -0,0 +1,98 @@
+/*
+ *
+ * Copyright (C) 2024 Intel Corporation
+ *
+ * Under the Apache License v2.0 with LLVM Exceptions. See LICENSE.TXT.
+ * SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+ *
+ */
+
+#include <assert.h>
+#include <hwloc.h>
+#include <stdlib.h>
+
+#include "base_alloc_global.h"
+#include "memory_target_numa.h"
+#include "memspace_internal.h"
+#include "memspace_numa.h"
+
+static umf_result_t umfMemspaceHostCreate(umf_memspace_handle_t *hMemspace) {
+    if (!hMemspace) {
+        return UMF_RESULT_ERROR_INVALID_ARGUMENT;
+    }
+
+    umf_result_t umf_ret = UMF_RESULT_SUCCESS;
+
+    hwloc_topology_t topology;
+    if (hwloc_topology_init(&topology)) {
+        // TODO: What would be an approrpiate err?
+        return UMF_RESULT_ERROR_UNKNOWN;
+    }
+
+    if (hwloc_topology_load(topology)) {
+        umf_ret = UMF_RESULT_ERROR_UNKNOWN;
+        goto err_topology_destroy;
+    }
+
+    // Shouldn't return -1, as 'HWLOC_OBJ_NUMANODE' deosn't appear to be an
```

Comment:
pehaps add and assert for that?

Reply:
Done, but I've set the `UMF_RESULT_ERROR_UNKNOWN` return value instead of assert.